TeamBuildingGame/Assets/Scripts/tcpclient/TCPClient.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Net;
+using System.Net.Sockets;
+using System;
+
+public class TCPClient : MonoBehaviour
+{
+    private TcpClient tcpClient;
+    private string ipAddress = "127.0.0.1";
+    private int portNumber = 10000;
+    private Vector2[] goalPositions;
+    private GoalZoneController goalZoneControllerScript;
+    private FieldGenerator fieldGeneratorScript;
+
+    public GameObject GoalZoneController;
+    public GameObject PlayingField;
+
+
+
+    void Awake()
+    {
+        // Establishes a udp connection on the port.
+        tcpClient = new TcpClient(ipAddress, portNumber);
+    }
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        goalPositions = new Vector2[2];
+        goalZoneControllerScript = GoalZoneController.GetComponent<GoalZoneController>();
+        fieldGeneratorScript = PlayingField.GetComponent<FieldGenerator>();
+
+        StartListening();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+
+    }
+
+    void OnDestroy()
+    {
+        // Ensures connection is closed when the game object it is attached to is destroyed, so it wont continue to receive messages after the game ends.
+        tcpClient.Close();
+    }
+
+    /// <summary> Receives the datagram. 
+    /// <param name="res">The result associated with the callback - the data.</param>
+    private void Receive(IAsyncResult res)
+    {
+
+    }
+
+    private void StartListening()
+    {
+        if (!tcpClient.Connected)
+        {
+            tcpClient.Connect(ipAddress, portNumber);
+        }
+        Debug.Log("Connected");
+        
+        NetworkStream stream = tcpClient.GetStream();
+
+        byte[] data = new byte[256];
```

Comment:
```suggestion
        byte[] data = new byte[1024];
```

det skal matche hosten